sci-libs/vtk/vtk-9.0.1.ebuild

```diff
@@ -0,0 +1,546 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+# TODO:
+# - add USE flag for remote modules? Those modules can be downloaded
+#	properly before building.
+
+PYTHON_COMPAT=( python3_{7,8,9} )
+WEBAPP_OPTIONAL=yes
+WEBAPP_MANUAL_SLOT=yes
+
+inherit check-reqs cmake cuda flag-o-matic java-pkg-opt-2 python-single-r1 toolchain-funcs virtualx webapp
+
+# Short package version
+MY_PV="$(ver_cut 1-2)"
+
+# needed for testing
+MISSING_TEST_FILES=(
+	eaab2cdba68e8630f9a1f8a8e81cb097
+	325acbb1a74ae27148d76399c2e7c7c5
+	8e09d028a9fd114b4ceb8f924d8b466c
+	7e7e6c5e28fa93b16fe0414881bdbbcd
+)
+
+DESCRIPTION="The Visualization Toolkit"
+HOMEPAGE="https://www.vtk.org/"
+SRC_URI="
+	https://www.vtk.org/files/release/${MY_PV}/VTK-${PV}.tar.gz
+	https://www.vtk.org/files/release/${MY_PV}/VTKData-${PV}.tar.gz
+	doc? ( https://www.vtk.org/files/release/${MY_PV}/vtkDocHtml-${PV}.tar.gz )
+	examples? ( https://www.vtk.org/files/release/${MY_PV}/VTKLargeData-${PV}.tar.gz )
+	test? (
+		https://www.vtk.org/files/ExternalData/MD5/eaab2cdba68e8630f9a1f8a8e81cb097
+		https://www.vtk.org/files/ExternalData/MD5/325acbb1a74ae27148d76399c2e7c7c5
+		https://www.vtk.org/files/ExternalData/MD5/8e09d028a9fd114b4ceb8f924d8b466c
+		https://www.vtk.org/files/ExternalData/MD5/7e7e6c5e28fa93b16fe0414881bdbbcd
+		https://www.vtk.org/files/release/${MY_PV}/VTKLargeData-${PV}.tar.gz
+	)
+"
+
+LICENSE="BSD LGPL-2"
+SLOT="0"
+# TODO: several dependencies need to be keyworded for arm
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+# TODO: New in 9.0?: sdl -> from which group / module is it coming?
+# Note: external xdmf2 has no recognized targets
+IUSE="+X all-modules boost cuda doc +eigen examples ffmpeg gdal imaging
+	java +json kits mpi mysql odbc offscreen openmp pegtl postgres python
+	qt5 +rendering tbb theora tk video_cards_nvidia views web"
+
+# most tests don't work
+RESTRICT="test"
+
+REQUIRED_USE="
+	all-modules? ( boost ffmpeg gdal mysql odbc postgres theora )
+	cuda? ( X )
+	python? ( ${PYTHON_REQUIRED_USE} )
+	qt5? ( X rendering )
+	tk? ( X rendering python )
+	web? ( python )
+"
+
+RDEPEND="
+	app-arch/lz4
+	app-arch/xz-utils
+	dev-db/sqlite
+	dev-cpp/eigen[cuda?,openmp?]
+	dev-libs/double-conversion:=
+	dev-libs/expat
+	dev-libs/icu:=
+	dev-libs/libxml2:2
+	dev-libs/pugixml
+	media-libs/libogg
+	media-libs/libpng
+	media-libs/libtheora
+	media-libs/tiff
+	sci-libs/hdf5:=[mpi=]
+	sci-libs/netcdf:=[mpi=]
+	sys-libs/zlib
+	virtual/jpeg
+	all-modules? ( sci-geosciences/liblas[gdal] )
+	boost? ( dev-libs/boost:=[mpi?] )
+	cuda? ( dev-util/nvidia-cuda-toolkit:= )
+	ffmpeg? ( media-video/ffmpeg:= )
+	gdal? ( sci-libs/gdal:= )
+	java? ( >=virtual/jdk-1.8:* )
+	json? ( dev-libs/jsoncpp:= )
+	mpi? (
+		sci-libs/h5part
+		sys-cluster/openmpi[cxx,romio]
+		virtual/mpi[cxx,romio]
```

Comment:
Specifying both `sys-cluster/openmpi` and `virtual/mpi` does not make sense.
`virtual/mpi` depends on `|| ( sys-cluster/openmpi sys-cluster/mpich sys-cluster/mpich2 ... )`.
If `sci-libs/vtk` works with any MPI implementation, then only `virtual/mpi` should be specified in dependencies.
If `sci-libs/vtk` works with only `sys-cluster/openmpi`, then only `sys-cluster/openmpi` should be specified in dependencies.

Reply:
I can't test this, because I can't get mpich to work. Gonna remove the `virtual/mpi` then.